---
layout: default
date: 2020-03-23
---

# A Report About Clinical Genome(ClinGen) Resource

_23 Mar 2020_

> [Report: Clinical Genome (ClinGen) Resource - Erin Ramos and Sharon Plon](https://www.youtube.com/watch?v=VZghJ95Qfzo)


## Why we want the ClinGen Resource

* 检测突变的能力大大超过将这些突变与临床影响建立解释的能力
* often without common data standards and approaches for interpretation
* Clinical Testing Lab Databases: Largely absent from public domain

### The ClinGen Program

Increase data sharing and build an authoritative resource to define the cilnical relevance of genes and variants for use in medicine and research.

* Launched: Sept 2013
* Phase II: Sept 2017

### Ecosystem

Buliding a Genomic Knowledge Base:

* Clinical Validity: Is this gene associated with a disease?
* Pathogenicity: Is this variant causative?
* Clinical Utility: Is this information actionable?

Shared Genetic and Helath Data by:

* Patients
* Clinicians
* Laboratories
* Researchers

-> Improve Patient Care Through Genomic Medicine

## Clinical Validity

> ClinGen developed semi-quantitative framework to classify strength of evidence for the role of genes in disease

Curated by:

* Genetic Evidence
  * Case-level
  * family segregation
  * case-control data
* Experimental Evidence
  * Expression
  * model organism
  * rescue studies
  * etc.

Classification:

Definitive -> Strong -> Moderate -> Limited -> No Evidence Reported -> ...

## Pathogenicity

### Muti-pronged effort needed for variant curation and interpretation

* Public sharing of variant interpretations via ClinVar
* Inter-laboratory conflict resolution
* Engaging experts in systematic consensus driven interpretation of variants (Expert Panels)
* Sequence Variants and Copy Number Variants

### ACMG standards and guidelines

Standards and guidelines for the interpretation of sequence variants: a joint consensus recommentdation of the American College of Medical Genetics and Genomics and the Association for Molecular Pathology

### Sequence Variant Interpretation WorkGroup

* refinethe ACMG/AMP guideline as they are deployed by the community
* move towards a more quantitative framework

i.e papers:

* Recommendations for interpreting the loss of function PVS1 ACMG/AMP variant criterion
* Modeling the ACMG/AMP variant classification guidelines as a Bayesian classification framework

### Others

* Variant Curation Interface
* CliGen Allele Registry
  * provides identifiers (CAID)
* CliGen's Variant Curation Expert Panels Progress
  * Define WG and plans
  * Develop Variant Classification rules
  * ...
* FDA Recognized Genetic Variant Database
* GlinGen is a GA4GH Driver Project

## Clinical Utility

ClinGen developed a framework to provide a transparent and systematic evidence base for prioritizing genes based on their clinical actionability.

### Clinical Actionability

* Well established clinical interventions
* Specific to the genetic disorder under consideration
* Lead to disease prevention or delayed onset, lowered clinical burden, or improved clinical outcomes

### Classification Matrix

Severity -> Likelihood -> Effectiveness of Intervention -> Nature of Intervention

## How engaging patients

### GenomeConnect

* Patient portal to engage patients in data sharing
  * Collects patient-entrted health information and genetic data extracted from genetic test reports
  * Transfers data into secure ClinGen-hosted environment
  * Connects patients with researchers and with other patients with the same condition

### Others

* Ancestry and Diversity WorkingGroup
* Engage and Train the Broader Community

---

# Population Data

> [Applying ACMG/AMP population data criteria for variants with low allele frequency and homozygous/hemizygous occurrences](https://www.youtube.com/watch?v=HWChT4ZIIgY)

## Overview

* Population data in the context of ACMG/AMP criteria
  * Variants with low allele frequence
  * Considerations regarding zygosity
* Both pathogenic and benign evidence

## Some key point

* Presence in gnomAD $\ne$ benign
* Absence in gnomAD $\ne$ pathoegnic




